app/scripts/controllers/projectBrowseCatalog.js

```diff
@@ -0,0 +1,49 @@
+'use strict';
+
+angular.module('openshiftConsole')
+  .controller('ProjectBrowseCatalogController',
+    function($scope,
+             $routeParams,
+             DataService,
+             APIService,
+             AuthorizationService,
+             Catalog,
+             CatalogService,
+             Navigate,
+             NotificationsService,
+             ProjectsService) {
+      var currentProjectName = $routeParams.project;
+      var imageStreamsVersion = APIService.getPreferredVersion('imagestreams');
+
+
+      ProjectsService.get(currentProjectName).then(function() {
+        AuthorizationService.getProjectRules(currentProjectName).then(function () {
+          var canIAddToProject = AuthorizationService.canIAddToProject(currentProjectName);
+          if (!canIAddToProject) {
+            Navigate.toProjectOverview(currentProjectName);
+            return;
+          }
+
+          // Show all catalog items, project templates, and project image streams
+          CatalogService.getCatalogItems().then(function (items) {
+            Catalog.getProjectCatalogItems(currentProjectName).then( _.spread(function(catalogServiceItems, errorMessage) {
+              DataService.list(imageStreamsVersion, {namespace: currentProjectName}, function(isData) {
```

Comment:
I think we can remove this call? Aren't we fetching it twice now since `getProjectCatalogItems` also request image streams?

Reply:
nit: We could request these in parallel and use `$q.all` so the page load slightly faster.